modules/claims_api/app/controllers/claims_api/v2/veterans/claims_controller.rb

```diff
@@ -210,21 +210,58 @@ def map_yes_no_to_boolean(key, value)
           end
         end
 
-        def map_bgs_tracked_items(bgs_claim)
+        def map_bgs_tracked_items(bgs_claim) # rubocop:disable Metrics/MethodLength
           return [] if bgs_claim.nil?
 
           claim_id = bgs_claim.dig(:benefit_claim_details_dto, :benefit_claim_id)
           return [] if claim_id.nil?
 
-          bgs_response = bgs_service.tracked_items.find_tracked_items(claim_id)
-          bgs = bgs_response.dig(:benefit_claim, :dvlpmt_items) || []
-          bgs.map do |item|
+          tracked_items = bgs_service
+                          .tracked_items
+                          .find_tracked_items(claim_id)
+                          .dig(:benefit_claim, :dvlpmt_items) || []
+          ebenefits_details = bgs_service
+                              .ebenefits_benefit_claims_status
+                              .find_benefit_claim_details_by_benefit_claim_id(
+                                benefit_claim_id: claim_id
+                              )[:benefit_claim_details_dto] || []
+
+          # Just in case there's a doc in ebenefits that we don't have a tracked_item for
+          ids = tracked_items.pluck(:dvlpmt_item_id)
+          docs = ((ebenefits_details[:wwsnfy] || []) + (ebenefits_details[:wwr] || []) + (ebenefits_details[:wwd] || []))
+          ids += docs.pluck(:dvlpmt_item_id)
+          ids = ids.uniq
+
+          ids.map.with_index do |id, i|
+            item = tracked_items.find { |t| t[:dvlpmt_item_id] == id } || {}
+            detail = docs.find do |doc|
+              doc[:dvlpmt_item_id] == id
+            end || {}
+
+            # Values for status enum: "NEEDED",
+            #                         "SUBMITTED_AWAITING_REVIEW",
+            #                         "INITIAL_REVIEW_COMPLETE",
+            #                         "ACCEPTED",
+            #                         "NO_LONGER_REQUIRED"
+            status = 'SUBMITTED_AWAITING_REVIEW' unless detail[:date_rcvd].nil?
+            status = 'NEEDED' if detail[:date_rcvd].nil?
+
+            uploads_allowed = ['NEEDED", "SUBMITTED_AWAITING_REVIEW", "INITIAL_REVIEW_COMPLETE']
```

Comment:
Is it ever possible for `status` to be `INITIAL_REVIEW_COMPLETE`?
Looking at this code, it looks like it could only ever be `SUBMITTED_AWAITIGN_REVIEW` or `NEEDED`.  🤔

Reply:
What did you determine for the `status`?

Reply:
I got statuses from [this comment](https://vajira.max.gov/browse/API-18165?focusedCommentId=1558349&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-1558349). Double check that I mapped the values correctly please.